Fetch video tracks

VincentCATILLON opened the pull request:
<!-- Please add Labels to the PR, and link it to Trello -->
## Detailed purpose of the PR

This PR uses tracks from state

### Result

![image](https://user-images.githubusercontent.com/8419208/71716448-17666980-2e15-11ea-99f3-4deeaa7a1a00.png)


### Testing strategy

- [x] I ensured my code is tested
- [ ] I took care to mention breaking changes and extra libs if appropriate

VincentCATILLON commented:
Should be tested with this code in canary: https://github.com/CoorpAcademy/components/pull/1656

VincentCATILLON commented on src/components/brand-theme-provider.stories.js:
```diff
@@ -21,15 +21,7 @@ const brand = createBrand({});
-          <Text>Logo: {brandTheme.images['logo-mobile']}</Text>
-        </React.Fragment>
-      )}
+      {brandTheme => <Text>{JSON.stringify(brandTheme, null, '\t')}</Text>}
```
This thing, is to show all stuff from brand

Volubyl replied:
c'est quoi le use case ?

VincentCATILLON replied:
To avoid editing this story everytime we are adding stuff in brand-theme context

Fausto95 replied:
Yay

VincentCATILLON commented on src/containers/video-controlable.js:
```diff
@@ -3,88 +3,85 @@
-      });
-      await this.props.fetchVideoUri(videoId, videoProvider);
-    }
+  currentTime: number | void;
```
from state to class property, because no render is needed

VincentCATILLON commented on src/containers/video-controlable.js:
```diff
@@ -186,21 +201,71 @@ class VideoControlable extends React.PureComponent<Props, State> {
+  source: getVideoSourceState(state, props),
+  tracks: getVideoTracksState(state, props),
+  selectedTrack: getVideoSelectedTrackState(state, props),
+  isFullScreen: getIsFullScreenState(state)
```
memoize on all connected props

VincentCATILLON commented on src/layer/data/index.js:
```diff
@@ -27,9 +27,9 @@ import {getCorrectAnswer} from './answers';
 import {logEvent} from './analytics';
 import {logError, setProperties as setLoggerProperties} from './logger';
-import {fetchLanguage, setLanguage, getInterfaceLanguage} from './language';
+import {setLanguage, getInterfaceLanguage} from './language';
```
Language fetch (datalayer, service and redux actions) has been removed because we already got the data after brand config fetch.

2 calls => 1 call

VincentCATILLON commented on src/layer/data/videos.js:
```diff
@@ -32,3 +33,28 @@ export const findUriById = async (id: string, provider: VideoProvider): Promise<
   return url;
 };
+
+export const findTracksById = async (
```
The stuff you wanna see

VincentCATILLON commented on src/modules/language.js:
```diff
@@ -0,0 +1,19 @@
+
+import {DEFAULT_LANGUAGE, SUPPORTED_LANGUAGES} from '../translations';
+
+export const getMatchingLanguage = (
```
Mutualized code to resolve matching language from a list of languages provided.
Used in two places:
- in authentication sign in action, when the brand has been fetched and we have to resolve and set the UI language
- when we got video tracks and wanna resolve the video track to select (and display)

VincentCATILLON commented on src/redux/actions/authentication.js:
```diff
@@ -95,13 +92,14 @@ export const signIn = (
-    const {services} = options;
+    const {supportedLanguages, defaultLanguage} = brand;
+    const deviceLanguage = services.Language.getFromInterface();
+    const language = getMatchingLanguage(supportedLanguages, defaultLanguage, deviceLanguage);
```
The thing you wanna see

Volubyl commented on src/__fixtures__/videos.js:
```diff
@@ -1,14 +1,38 @@
 import {VIDEO_PROVIDER} from '../layer/data/_const';
 
-export const createVideoUrl = (id: string, provider: VideoProvider): string => {
+export const createVideoUri = (id: string, provider: VideoProvider): string => {
```
uri url ?

VincentCATILLON replied:
URI everywhere in this case because i had to chose between uri and url in player store and there was the 2 things

Volubyl commented on src/components/lesson.js:
```diff
@@ -15,7 +14,6 @@ import QuestionTitle from './question-title';
 import Resource from './resource';
 import ResourcesBrowser from './resources-browser';
 import Space from './space';
-import {BrandThemeContext} from './brand-theme-provider';
```
qu'est ce qui explique que l'on n'a plus besoin du brandThemeProvider ?

VincentCATILLON replied:
Because we don't need anymore the brand host as it's constructed by the container and on demand

Volubyl commented:
j'ai pas trop compris le truc que tu as changé autour des langues tu peux expliquer ?

VincentCATILLON commented:
@Volubyl

The first of all, I extracted some code from the language service to be able to use it elsewhere, like in the containers `video-controlable` to have the same logic to resolve language to use (depending on platform language) and to resolve the language to display for the tracks.

From there, I spotted that the brand config was fetched twice, so I removed the language fetch actions (+ service and data layer so), and the `set` action around language, was overkill so I removed it to only have a service (like analytics, logger, etc).

VincentCATILLON commented:
Functional test on Android made by @Volubyl: ✅ 
Functional test on iOS made by me: ✅
